src/NuGet.Core/Microsoft.Build.NuGetSdkResolver/NuGetSdkResolver.cs

```diff
@@ -73,14 +73,14 @@ public override SdkResult Resolve(SdkReference sdkReference, SdkResolverContext
             // Escape hatch to disable this resolver
             if (DisableNuGetSdkResolver.Value)
             {
-                return factory.IndicateFailure(errors: null, warnings: new List<string>() { Strings.Error_DisabledSdkResolver });
+                return factory.IndicateFailure(errors: new List<string>() { Strings.Error_DisabledSdkResolver }, warnings: null);
```

Comment:
`DisableNuGetSdkResolver` is true when the customer sets an environment variable to opt-in to this behaviour (to opt-out of the feature). By making this scenario an error, it completely breaks the opt-out feature.  Is this intentional?  If so, would it be better to just delete `DisableNuGetSdkResolver`?

Reply:
In this case it won't actually "break" anything and no errors are physically logged by the SDK resolver.  Only a list of errors will be returned to MSBuild.  The logic in MSBuild will then move on to the next SDK resolver and if none of them work then _all_ of the errors from _all_ of the SDK resolvers are logged, as well as all of the warnings.  

If you opt out of using the NuGet-based MSBuild project SDK resolver and another SDK resolver is able to resolve the SDK, then these errors returned to MSBuild are never logged.  Clear as mud right?  So in the case where the NuGet SDK resolver is "failing" because you disabled it, you'll get an error if no SDK resolver resolves it so you at least know that you might have forgotten that you disabled it.  But you'll also see errors from the other SDK resolvers so maybe you just misspelled the SDK name or something.

https://github.com/dotnet/msbuild/blob/main/src/Build/BackEnd/Components/SdkResolution/SdkResolverService.cs#L227

Reply:
In this case, I think a comment is warranted, because this is totally non-obvious and completely contradictory to how errors vs warnings work in restore, pack, basically all other parts of nuget.

Reply:
Great idea, I'll add a comment here.

Reply:
Thanks! I agree that how MSBuild is handling these is a bit confusing.